### ISSUE TITLE
Parse content from docstrings?

### ISSUE THREAD

**Issue by Cadair** [cjw296/sybil#25]
I am looking sybil as a replacement for the built in doctest collector in pytest, so I can more easily extend the functionality with setup code blocks and namespaces, all the good things sybil supports. 

The one thing I am not clear about is if sybil can parse blocks from docstrings inside the python source? Which is something the doctest collector can do.

**Comment by cjw296** [comment]
It can't currently, but I'd like it to be able to.

Is your code publicly accessible? I'd like to take a look at your current docstring tests and how you have pytest configured to find them, then hopefully I can see the best way for Sybil to implement this :-)

**Comment by Cadair** [comment]
yeah, the code is https://github.com/sunpy/sunpy/ and https://github.com/astropy/astropy currently we use https://github.com/astropy/pytest-doctestplus to give us a little extra control over the running of the tests.

**Comment by cjw296** [comment]
Ah, interesting, pytest-doctestplus looks like an alternative to sybil, so curious what you're looking for that pytest-doctestplus doesn't currently offer but sybil does?

**Comment by Cadair** [comment]
If I remember correctly, I think I was looking at the possibility of refactoring pytest-doctestplus to use sybil as it's parser to extract the test and have setup and teardown code etc. I spent a morning months ago falling down this rabbit hole and now can't exactly remember why lol

**Comment by cjw296** [comment]
Just to note: this is something I'd love to see, but unsure of the best way to go about implementing.
